### ISSUE TITLE
Scrollbar appears when column resizing is enabled

### ISSUE THREAD

**Issue by williamyeny** [KevinVandy/mantine-react-table#32]
### mantine-react-table version

v0.9.5

### react & react-dom versions

v17

### Describe the bug and the steps to reproduce it

With column resizing enabled, a scrollbar appears on the bottom. This appears to be from [MRT_TableHeadCellResizeHandle](https://github.com/KevinVandy/mantine-react-table/blob/8b0389164ae39c1851776c78238da8d2af6bcf90/packages/mantine-react-table/src/head/MRT_TableHeadCellResizeHandle.tsx) overflowing the header. I am using Chrome Version 111.0.5563.146.

### Minimal, Reproducible Example - (Optional, but Recommended)

Visible in the official docs: https://www.mantine-react-table.com/docs/guides/column-resizing#column-resize-mode

### Screenshots or Videos (Optional)

![image](https://user-images.githubusercontent.com/12968964/230452898-582ed5bb-a2f2-4fc3-a141-b68ef69483ac.png)

### Do you intend to try to help solve this bug with your own PR?

Maybe, I'll investigate and start debugging

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by williamyeny** [comment]
I'm able to solve this by going into the `<Box />` of MRT_TableHeadCellResizeHandle, removing the right padding, and adding 4 px to the right margin. The tradeoff is that the click area will be reduced. If this tradeoff is not worth it to be merged, is it possible to implement this style in my project?

**Comment by Guria** [comment]
@williamyeny 
Following workaround works for me:
```
mantineTableHeadCellProps={{
  sx: {
    '& .mantine-TableHeadCell-Content': {
      transform: 'translateX(-4px)',
      paddingLeft: 4,
    },
  },
}}
```

**Comment by KevinVandy** [comment]
fixed in v1.0.0-beta.3
